app/uk/gov/hmrc/economiccrimelevyreturns/controllers/AmendReasonController.scala

```diff
@@ -55,12 +59,17 @@ class AmendReasonController @Inject() (
       .bindFromRequest()
       .fold(
         formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode, request.startAmendUrl))),
-        reason =>
-          eclReturnsConnector
-            .upsertReturn(request.eclReturn.copy(amendReason = Some(reason)))
-            .map { updatedReturn =>
-              Redirect(pageNavigator.nextPage(mode, updatedReturn))
-            }
+        reason => {
+          val updatedReturn = request.eclReturn.copy(amendReason = Some(reason))
+          (for {
+            unit <- eclReturnsService
+                      .upsertReturn(updatedReturn)
+                      .asResponseError
+          } yield updatedReturn).fold(
+            err => Status(err.code.statusCode)(Json.toJson(err)),
```

Comment:
How does this error message manifest? We should be showing the default error page, but this looks like it'll show JSON like what's happening in Account server.